hack/gke/xfs-format/xfs-format.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+ephemeral_mountpoint=/mnt/stateful_partition/kube-ephemeral-ssd
+raid_device=$( awk '$2=="/mnt/stateful_partition/kube-ephemeral-ssd" {print $1}' /proc/mounts )
+raid_dir=/mnt/raid-disks/disk0
+
+if [[ -z ${raid_device} ]]; then
+  echo "Raid device not mounted"
+  exit 1
+fi
+
+if [[ "$( blkid -o value -s TYPE ${raid_device} )" == "xfs" ]]; then
+  echo "Raid device's file system is already xfs."
+  exit 0
+fi
+
+systemctl stop kube-node-configuration.service
+systemctl stop kubelet.service
+/home/kubernetes/bin/crictl stopp $( /home/kubernetes/bin/crictl pods -q )
+systemctl stop containerd.service
+
+tmp_dir=$( mktemp -d )
+cp -af ${ephemeral_mountpoint}/. ${tmp_dir}/
+
+umount $( grep -e '/home/kubernetes/containerized_mounter' -e '/var/lib/containerd' -e '/home/containerd' /proc/mounts | sed -E -e 's/([^ ]+) ([^ ]+) .+/\2/' | sort -u -r ) || true
+# Separate line due to existing dependencies
+umount $( grep -e "${raid_device}" /proc/mounts | sed -E -e 's/([^ ]+) ([^ ]+) .+/\2/' | sort -r ) || true
+mkfs.xfs -f ${raid_device}
+
+cp -af ${tmp_dir}/. ${ephemeral_mountpoint}/
+rm -rf ${tmp_dir}
+
+mkdir -p ${ephemeral_mountpoint}/scylla
+mkdir -p ${raid_dir}
+chmod a+w ${raid_dir}
+
+reboot & disown
```

Comment:
`exec reboot`?

Reply:
I wanted to fork to avoid seeing `xfs-format.service: Main process exited, code=killed, status=15/TERM` in xfs-format's journal, I suppose its somewhat confusing. On the other hand, forking logs another start/stop of the service. Wdyt?

Reply:
I haven't used it but looking at `systemctl show systemd-reboot.service`,`SuccessAction=reboot` might work

Reply:
SuccessAction is not recognised with oneshot. I've used ExecStartPost=/usr/sbin/reboot which is only triggered if ExecStart ran successfully, so it required changing the status code to 1 if the device is already xfs and checking for that in the daemonset as well. The sigkill is logged in the journal anyway, but I guess we can live with it.
```
Sep 14 11:54:24 gke-rzetelskik-scylla-dem-scylla-pool-465fca7d-r468 xfs-format.sh[10568]: + chmod a+w /mnt/raid-disks/disk0
Sep 14 11:54:24 gke-rzetelskik-scylla-dem-scylla-pool-465fca7d-r468 systemd[1]: xfs-format.service: Control process exited, code=killed, status=15/TERM
```